ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -34,6 +34,42 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
         )
   }
 
+  object CommandErrors extends ErrorGroup() {
+
+    @Explanation("This rejection is given when a command submission could not be processed.")
+    @Resolution("Refer to the error details and/or retry.")
+    object CommandSubmissionFailure
+        extends ErrorCode(
+          id = "COMMAND_SUBMISSION_FAILURE",
+          ErrorCategory.ContentionOnSharedResources,
+        ) {
+      case class Reject(messagePrefix: String, override val throwableO: Option[Throwable])(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = {
+              val suffix: String =
+                throwableO.fold("")(t => s": ${t.getClass.getSimpleName}: ${t.getMessage}")
+              s"$messagePrefix$suffix"
+            }
+          )
+    }
+
+    @Explanation("This rejection is given when a command completion has failed.")
```

Comment:
It sounds rather strange for a completion to fail. Instead, a command can fail and the completion can signal that asynchronously.

Reply:
I think `DEADLINE_EXCEEDED/DeadlineExceededRequestStateUnknown`  is better than `ABORTED/ContentionOnSharedResources`.
We already have `LedgerApiErrors.WriteErrors.RequestTimeOut` for that so using it like so:
```
v2 = LedgerApiErrors.WriteErrors.RequestTimeOut.Reject("Timed out while awaiting for a completion corresponding to a command submission.", definiteAnswer = false).asGrpcStatusFromContext,
```
A concern is whether `WriteErrors` error grouping is appropriate for the case of completion timeouts?

Also,  `CommandCompletionFailure` is used only in one place so removing it.

Reply:
Let's not worry about error grouping in this PR. These are overhauled in https://github.com/digital-asset/daml/pull/11686